NewForce-at-Mountwest/bangazonapi-team-donkoss
# Ticket #2 -- Product Controller with Test

## Proposed changes

Product Controller allows the user to `GET` a single entry, and `GET`, `POST`, `PUT`, and `DELETE` entries from the Product table in the BangazonAPI Database. 

Included is the test module `ProductTest` that will test all methods automatically. 

## Related Issues

`Ticket #2 `

https://trello.com/c/xNcjlvgR

## Steps To Test

clone my [repository](https://github.com/NewForce-at-Mountwest/bangazonapi-team-donkoss/tree/af-ticket2)
`git fetch --a` to get access to my branch
`git checkout af-ticket2` switch to my branch
`git pull origin af-ticket2` to grab the latest commit

`start BangazonAPI.csproj` to open the project

create a new database and title it `BangazonAPI`

open a new SQL query in `SQL Object Explorer` and run this [script](https://github.com/NewForce-at-Mountwest/bangazonapi-team-donkoss/blob/61cd0b2c1aed09c6988c016f7dedeae777e0751c/BangazonAPI/dummyData.sql) to populate the database with dummy data

In the Solution Explorer, right click on `BangazonAPITest` and select `Run Tests`

The user should be able to `GET`, `POST`, `PUT`, and `DELETE` entries in the Product table in the BangazonAPI database using Postman.

The controller is set to "soft" delete entries from the database, while the test is set to "hard" delete the dummy data it creates.

## What kind of change does this PR introduce?
Please check one of the following:
